test/links/links_test.py

```diff
@@ -0,0 +1,122 @@
+
+    with open(markdown_path) as markdown_file:
+        content = markdown_file.read()
+        link_matches = re.findall(pattern_to_find_links, content)
+        for match in link_matches:
+            link = match[1]
+            pages[title]["links"].append(link)
+
+        anchor_matches = re.findall(pattern_to_find_anchors, content)
+        for match in anchor_matches:
+            anchor = match.replace(r"^[^a-zA-Z]+", "").replace(r" ", "-").lower()
+            anchor = re.sub("[^a-zA-Z0-9 -]+", "", anchor)
+            pages[title]["anchors"].append(anchor)
+
+        template_call_matches = re.findall(pattern_to_find_template_calls, content)
+        for template_call in template_call_matches:
+            # page contains liquid templates. find the anchors within the .yml file used by the liquid template
+            for attribute in template_call.split(" "):
+                if "data=" in attribute:
+                    yaml_file_path = attribute.split("=")[1]\
+                                         .replace("site.data.", "")\
+                                         .replace(".", "/")\
+                                         .replace("_", "-") + ".yml"
+                    # although the directory name is expected to have - instead of _
+                    # file name is expected to have _ , so:
+                    yaml_file_name = os.path.basename(yaml_file_path)
+                    new_yaml_file_name = yaml_file_name.replace("-", "_")
+                    yaml_file_path = yaml_file_path.replace(yaml_file_name, new_yaml_file_name)
+
+                    yaml_content = yaml.load(open(yaml_file_path), Loader=yaml.CLoader)
+                    # looks for the value of all keys named `title`, convert them to an id and add them as anchors
+                    for found_title in nested_lookup("title", yaml_content):
+                        anchor = found_title.replace(r" ", "-").replace(r"/", "-").lower()
+                        pages[title]["anchors"].append(anchor)
+
+
+class LinksTest(unittest.TestCase):
+
+        def test_native_links(self):
+            errors = []
+            for title, links_and_anchors in pages.items():
+                for link in links_and_anchors["links"]:
+                    link_page = link.split("#")[0]
+                    link_page = link_page.split("?")[0]  # removes params
+                    if link_page not in pages:
+                        errors.append("The link [" + link_page + "] found in [" + title + "] is broken.")
+
+                    link_has_anchor = len(link.split("#")) > 1
+                    if link_has_anchor:
+                        link_anchor = link.split("#")[1]
+                        if link_page in pages and link_anchor not in pages[link_page]["anchors"]:
+                            errors.append("The anchor [" + link_anchor + "] of the link [" + link_page + "] found in [" + title + "] is broken.")
+            print(errors)
```

Comment:
cleaner to put the error message in the `assertEqual`: `self.assertEqual(len(errors), 0, msg=errors)` (might need to do `str(errors)`